lib/data/external/AwsClient.js

```diff
@@ -7,6 +7,33 @@ const { prepareStream } = require('../../api/apiUtils/object/prepareStream');
 const { logHelper, removeQuotes, trimXMetaPrefix } = require('./utils');
 const { config } = require('../../Config');
 
+function _awsGetPreCheck(objectGetInfo, reqUids, callback) {
+    const log = createLogger(reqUids);
+    const key = typeof objectGetInfo === 'string' ? objectGetInfo :
+        objectGetInfo.key;
+    const versionId = typeof objectGetInfo === 'string' ? undefined :
+        objectGetInfo.dataStoreVersionId;
+    return this._client.headObject({
+        Bucket: this._awsBucketName,
+        Key: key,
+        VersionId: versionId,
+    }, err => {
+        if (err) {
+            logHelper(log, 'error', 'error heading object ' +
+            'from datastore', err, this._dataStoreName);
+            if (err.code === 'NotFound') {
+                const error = errors.InternalError.customizeDescription(
+                    'Unexpected error from AWS: "NoSuchVersion". Backend ' +
+                    'data may have diverged from our metadata records.'
```

Comment:
why NoSuchVersion instead of NotFound? Also, I would do something like 'Data on AWS may have been altered outside of CloudServer.'

Reply:
updated